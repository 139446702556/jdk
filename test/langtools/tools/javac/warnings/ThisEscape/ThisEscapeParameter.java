/*
 * @test /nodynamiccopyright/
 * @bug 8015831
 * @compile/ref=ThisEscapeParameter.out -Xlint:this-escape -XDrawDiagnostics ThisEscapeParameter.java
 * @summary Verify 'this' escape detection handles leaks via passing 'this' as a parameter
 */

public class ThisEscapeParameter {

    public ThisEscapeParameter() {
        ThisEscapeParameter.method(this);
    }

    public static void method(Object obj) {
        obj.hashCode();
    }
}
