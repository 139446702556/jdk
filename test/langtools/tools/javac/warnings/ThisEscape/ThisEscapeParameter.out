ThisEscapeParameter.java:15:21: compiler.warn.possible.this.escape
ThisEscapeParameter.java:11:35: compiler.warn.possible.this.escape.location
2 warnings
